homeassistant/components/ssdp/__init__.py

```diff
@@ -56,6 +56,7 @@
 ATTR_UPNP_UPC = "UPC"
 ATTR_UPNP_PRESENTATION_URL = "presentationURL"
 
+PRIMARY_MATCH_KEYS = [ATTR_UPNP_MANUFACTURER, "st", ATTR_UPNP_DEVICE_TYPE]
```

Comment:
Should `st` be used? `st` is only available in search responses, not in advertisements. The callback receives the device/service type `dts` in `_ssdp_listener_callback` which is always set whether the device is seen via a search or an advertisement. Should `dts` be added to the `combined_headers` dict, and components can match on that?

Reply:
I think we have to use `st` since its already in the `manifest.json`s

```
% grep -A 4 'ssdp' */manifest.json
arcam_fmj/manifest.json:  "ssdp": [
arcam_fmj/manifest.json-    {
arcam_fmj/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:MediaRenderer:1",
arcam_fmj/manifest.json-      "manufacturer": "ARCAM"
arcam_fmj/manifest.json-    }
--
axis/manifest.json:  "ssdp": [
axis/manifest.json-    {
axis/manifest.json-      "manufacturer": "AXIS"
axis/manifest.json-    }
axis/manifest.json-  ],
--
control4/manifest.json:  "ssdp": [
control4/manifest.json-    {
control4/manifest.json-      "st": "c4:director"
control4/manifest.json-    }
control4/manifest.json-  ],
--
deconz/manifest.json:  "ssdp": [
deconz/manifest.json-    {
deconz/manifest.json-      "manufacturer": "Royal Philips Electronics"
deconz/manifest.json-    }
deconz/manifest.json-  ],
--
denonavr/manifest.json:  "ssdp": [
denonavr/manifest.json-    {
denonavr/manifest.json-      "manufacturer": "Denon",
denonavr/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:MediaRenderer:1"
denonavr/manifest.json-    },
--
directv/manifest.json:  "ssdp": [
directv/manifest.json-    {
directv/manifest.json-      "manufacturer": "DIRECTV",
directv/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:MediaServer:1"
directv/manifest.json-    }
--
dlna_dmr/manifest.json:  "dependencies": ["network", "ssdp"],
dlna_dmr/manifest.json:  "ssdp": [
dlna_dmr/manifest.json-    {
dlna_dmr/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:MediaRenderer:1"
dlna_dmr/manifest.json-    },
dlna_dmr/manifest.json-    {
--
fritz/manifest.json:  "ssdp": [
fritz/manifest.json-    {
fritz/manifest.json-      "st": "urn:schemas-upnp-org:device:fritzbox:1"
fritz/manifest.json-    }
fritz/manifest.json-  ],
--
fritzbox/manifest.json:  "ssdp": [
fritzbox/manifest.json-    {
fritzbox/manifest.json-      "st": "urn:schemas-upnp-org:device:fritzbox:1"
fritzbox/manifest.json-    }
fritzbox/manifest.json-  ],
--
harmony/manifest.json:  "ssdp": [
harmony/manifest.json-    {
harmony/manifest.json-      "manufacturer": "Logitech",
harmony/manifest.json-      "deviceType": "urn:myharmony-com:device:harmony:1"
harmony/manifest.json-    }
--
heos/manifest.json:  "ssdp": [
heos/manifest.json-    {
heos/manifest.json-      "st": "urn:schemas-denon-com:device:ACT-Denon:1"
heos/manifest.json-    }
heos/manifest.json-  ],
--
huawei_lte/manifest.json:  "ssdp": [
huawei_lte/manifest.json-    {
huawei_lte/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:InternetGatewayDevice:1",
huawei_lte/manifest.json-      "manufacturer": "Huawei"
huawei_lte/manifest.json-    }
--
hue/manifest.json:  "ssdp": [
hue/manifest.json-    {
hue/manifest.json-      "manufacturer": "Royal Philips Electronics",
hue/manifest.json-      "modelName": "Philips hue bridge 2012"
hue/manifest.json-    },
--
hyperion/manifest.json:  "ssdp": [
hyperion/manifest.json-    {
hyperion/manifest.json-      "manufacturer": "Hyperion Open Source Ambient Lighting",
hyperion/manifest.json-      "st": "urn:hyperion-project.org:device:basic:1"
hyperion/manifest.json-    }
--
isy994/manifest.json:  "ssdp": [
isy994/manifest.json-    {
isy994/manifest.json-      "manufacturer": "Universal Devices Inc.",
isy994/manifest.json-      "deviceType": "urn:udi-com:device:X_Insteon_Lighting_Device:1"
isy994/manifest.json-    }
--
keenetic_ndms2/manifest.json:  "ssdp": [
keenetic_ndms2/manifest.json-    {
keenetic_ndms2/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:InternetGatewayDevice:1",
keenetic_ndms2/manifest.json-      "manufacturer": "Keenetic Ltd."
keenetic_ndms2/manifest.json-    },
--
konnected/manifest.json:  "ssdp": [
konnected/manifest.json-    {
konnected/manifest.json-      "manufacturer": "konnected.io"
konnected/manifest.json-    }
konnected/manifest.json-  ],
--
netgear/manifest.json:  "ssdp": [
netgear/manifest.json-    {
netgear/manifest.json-      "manufacturer": "NETGEAR, Inc.",
netgear/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:InternetGatewayDevice:1"
netgear/manifest.json-    }
--
roku/manifest.json:  "ssdp": [
roku/manifest.json-    {
roku/manifest.json-      "st": "roku:ecp",
roku/manifest.json-      "manufacturer": "Roku",
roku/manifest.json-      "deviceType": "urn:roku-com:device:player:1-0"
--
samsungtv/manifest.json:  "ssdp": [
samsungtv/manifest.json-    {
samsungtv/manifest.json-      "st": "urn:samsung.com:device:RemoteControlReceiver:1"
samsungtv/manifest.json-    }
samsungtv/manifest.json-  ],
--
songpal/manifest.json:  "ssdp": [
songpal/manifest.json-    {
songpal/manifest.json-      "st": "urn:schemas-sony-com:service:ScalarWebAPI:1",
songpal/manifest.json-      "manufacturer": "Sony Corporation"
songpal/manifest.json-    }
--
sonos/manifest.json:  "dependencies": ["ssdp"],
sonos/manifest.json-  "after_dependencies": ["plex", "zeroconf"],
sonos/manifest.json-  "zeroconf": ["_sonos._tcp.local."],
sonos/manifest.json:  "ssdp": [
sonos/manifest.json-    {
sonos/manifest.json-      "st": "urn:schemas-upnp-org:device:ZonePlayer:1"
sonos/manifest.json-    }
sonos/manifest.json-  ],
--
syncthru/manifest.json:  "ssdp": [
syncthru/manifest.json-    {
syncthru/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:Printer:1",
syncthru/manifest.json-      "manufacturer": "Samsung Electronics"
syncthru/manifest.json-    }
--
synology_dsm/manifest.json:  "ssdp": [
synology_dsm/manifest.json-    {
synology_dsm/manifest.json-      "manufacturer": "Synology",
synology_dsm/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:Basic:1"
synology_dsm/manifest.json-    }
--
unifi/manifest.json:  "ssdp": [
unifi/manifest.json-    {
unifi/manifest.json-      "manufacturer": "Ubiquiti Networks",
unifi/manifest.json-      "deviceType": "urn:schemas-upnp-org:device:InternetGatewayDevice:1",
unifi/manifest.json-      "modelDescription": "UniFi Dream Machine"
--
upnp/manifest.json:  "ssdp": [
upnp/manifest.json-    {
upnp/manifest.json-      "st": "urn:schemas-upnp-org:device:InternetGatewayDevice:1"
upnp/manifest.json-    },
upnp/manifest.json-    {
--
wemo/manifest.json:  "ssdp": [
wemo/manifest.json-    {
wemo/manifest.json-      "manufacturer": "Belkin International Inc."
wemo/manifest.json-    }
wemo/manifest.json-  ],
--
wilight/manifest.json:  "ssdp": [
wilight/manifest.json-    {
wilight/manifest.json-      "manufacturer": "All Automacao Ltda"
wilight/manifest.json-    }
wilight/manifest.json-  ],
--
yamaha_musiccast/manifest.json:  "ssdp": [
yamaha_musiccast/manifest.json-    {
yamaha_musiccast/manifest.json-      "manufacturer": "Yamaha Corporation"
yamaha_musiccast/manifest.json-    }
yamaha_musiccast/manifest.json-  ],
--
```

Reply:
Alright, works for me.